File: openapi/code_samples/csharp/api_company_sector_{caen}_average.cs
using Microsoft.AspNetCore.Mvc;
using Swashbuckle.AspNetCore.Annotations;

[ApiController]
[Route("api/[controller]")]
public class CompanyController : ControllerBase
{
    [HttpGet]
    [Route("sector-average/{caen}")]
    [SwaggerOperation(Summary = "Retrieves the company average sector on a given CAEN.")]
    [ProducesResponseType(typeof(SectorAverageView[]), 200)]
    [ProducesResponseType(typeof(Fault), 400)]
    public IActionResult GetSectorAverageByCaen([FromRoute] string caen)
    {
        // Your implementation here
        return Ok();
    }
}

File: openapi/code_samples/python/api_company_sector_{caen}_average.py
from flask import Flask, request, jsonify

app = Flask(__name__)

@app.route('/api/company/sector-average/<string:caen>', methods=['GET'])
def get_sector_average_by_caen(caen):
    # Your implementation here
    return jsonify({}), 200

if __name__ == '__main__':
    app.run()
